openlibrary/plugins/openlibrary/js/image_change.js

```diff
@@ -0,0 +1,77 @@
+export function initChangeImageImport() {
+
+    // Pull data from data-config of class "manageCovers" in covers/manage.html
+    var data_array = $('.manageCovers').attr('data-config').split(',');
+    var doc_type_key = data_array[0];
+    var add_url = data_array[3];
+    var manage_url = data_array[4];
+
+    function add_iframe(selector, src) {
+        $(selector)
+            .append('<iframe frameborder="0" height="450" width="580" marginheight="0" marginwidth="0" scrolling="auto"></iframe>')
+            .find('iframe')
+            .attr('src', src);
+    }
+
+    // Add iframes lazily when the popup is loaded.
+    // This avoids fetching the iframes along with main page.
+    $('.coverPop')
+        .bind('click', function () {
+            // clear the content of #imagesAdd and #imagesManage before adding new
+            $('#imagesAdd').html('');
+            $('#imagesManage').html('');
+            if (doc_type_key === '/type/work') {
+                $('#imagesAdd').prepend('<div class="throbber"><h3>$_("Searching for covers")</h3></div>');
+            }
+            setTimeout(function () {
+                // add iframe to add images
+                add_iframe('#imagesAdd', add_url);
+                // add iframe to manage images
+                add_iframe('#imagesManage', manage_url);
+            }, 0);
+        })
+        .bind('cbox_cleanup', function () {
+            $('#imagesAdd').html('');
+            $('#imagesManage').html('');
+        });
+
+}
+
+export function updateImage(image) {
+    // Pull data from data-config of class "manageCovers" in covers/manage.html
+    var data_array = $('.manageCovers').attr('data-config').split(',');
+    var doc_type_key = data_array[0];
+    var coverstore_url = data_array[1];
+    var cover_selector = data_array[2];
+    var url;
+
+    if (['/type/edition', '/type/work', '/edit'].includes(doc_type_key)) {
+        if (image) {
+            url = `${coverstore_url}/b/id/${image}-M.jpg`;
```

Comment:
Looking at this line and L69 I'm wondering if data-config should just pass the image_url and contain this logic and use the value directly. Seems little reason to have to do this in JS.

Reply:
If the image being saved is for an author, the url differs from that of an edition or work. The logic to handle this would, I believe, have to be handled by web.py. I can make this change if you think it would be better, but I just left this in for now since the logic expressed in the JavaScript would essentially be the same as in the web.py solution.